perf/perf_stress_ng.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: kajol Jain<kjain@linux.ibm.com>
+#         Manvanthara Puttashakar <manvanth@linux.vnet.ibm.com>
+
+import os
+import re
+import psutil
+import time
+from avocado.utils import process
+from avocado import Test
+from avocado.utils import process, build, archive
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+def clear_dmesg():
+    process.run("dmesg -C ", sudo=True)
+
+
+def collect_dmesg(object):
+    return process.system_output("dmesg").decode()
+
+
+class Stressng(Test):
+
+    """
+    Perf record with CPU stress
+    :source: git://kernel.ubuntu.com/cking/stress-ng.git
+    :param cpu_per: CPU Percentage to load
+    """
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.timeout = self.params.get('timeout', default='1')
+        self.cpu_per = self.params.get("cpu_load", default='10')
+        self.profile_dur = int(
+            self.params.get(
+                "profile_duration",
+                default='1'))
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            asset_url = 'https://github.com/ColinIanKing/stress-ng/archive/master.zip'
+            tarball = self.fetch_asset(
+                'stressng.zip', locations=[asset_url], expire='7d')
+            archive.extract(tarball, self.workdir)
+            sourcedir = os.path.join(self.workdir, 'stress-ng-master')
+            os.chdir(sourcedir)
+            result = build.run_make(
+                sourcedir, process_kwargs={
+                    'ignore_status': True})
+            for line in str(result).splitlines():
+                if 'error:' in line:
+                    self.cancel(
+                        "Build Failed, Please check the build logs for details !!")
+            build.make(sourcedir, extra_args='install')
+            clear_dmesg()
+        elif run_type == "distro":
+            deps = ['gcc', 'perf', 'stress-ng']
```

Comment:
please append logic to install GCC and perf package when upstream as a target.

Reply:
Done resolved